CRM/Core/Error.php

```diff
@@ -586,6 +586,9 @@ public static function debug_log_message($message, $out = FALSE, $prefix = '', $
       if ($config->userSystem->is_drupal and function_exists('watchdog')) {
         watchdog('civicrm', '%message', ['%message' => $message], $priority ?? WATCHDOG_DEBUG);
       }
+      elseif ($config->userSystem->is_drupal and CIVICRM_UF == 'Drupal8') {
+        \Drupal::logger('civicrm')->debug($message);
```

Comment:
Thanks for the patch. This would always log with priority "debug". I think it should use the `log` function like `\Drupal::logger('civicrm')->log($priority ?? \Drupal\Core\Logger\RfcLogLevel::DEBUG, '%message', ['%message' => $message])`

The '%' part is just to make it the same as drupal 7. Could also be `\Drupal::logger('civicrm')->log($priority ?? \Drupal\Core\Logger\RfcLogLevel::DEBUG, $message, [])`.

I see the note on line 585. I'm not sure what the situation it talks about is but I guess it means you can't put this in the UF class instead so yes it has to go here.

Reply:
thanks @demeritcowboy , updated patch..